REGRESSION(Sequoia): [ macOS Sequoia wk2 Debug x86_64 ] webrtc/captureCanvas-webrtc-software-h264-high.html is a constant failure.

youennf opened the pull request:
#### 3d8e0be1db6ae9348e769adde95b7eff68897ff9
<pre>
REGRESSION(Sequoia): [ macOS Sequoia wk2 Debug x86_64 ] webrtc/captureCanvas-webrtc-software-h264-high.html is a constant failure.
<a href="https://rdar.apple.com/145695939">rdar://145695939</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=288655">https://bugs.webkit.org/show_bug.cgi?id=288655</a>

Reviewed by Philippe Normand.

We had bugs related to software H264 encoding on intel macOS and added related tests.
This test infrastructure was only available if H264 encoding was done within WebProcess, which is now an unsupported configuration.
We remove the test and corresponding obsolete test infrastructure.

* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/mac-site-isolation/TestExpectations:
* LayoutTests/platform/mac-wk2/TestExpectations:
* LayoutTests/platform/mac/TestExpectations:
* LayoutTests/platform/mac/webrtc/captureCanvas-webrtc-software-encoder-expected.txt: Removed.
* LayoutTests/platform/mac/webrtc/captureCanvas-webrtc-software-encoder.html: Removed.
* LayoutTests/platform/wpe/TestExpectations:
* LayoutTests/webrtc/captureCanvas-webrtc-software-h264-baseline-expected.txt: Removed.
* LayoutTests/webrtc/captureCanvas-webrtc-software-h264-baseline.html: Removed.
* LayoutTests/webrtc/captureCanvas-webrtc-software-h264-high-expected.txt: Removed.
* LayoutTests/webrtc/captureCanvas-webrtc-software-h264-high.html: Removed.
* Source/ThirdParty/libwebrtc/Configurations/libwebrtc.exp:
* Source/ThirdParty/libwebrtc/Source/webrtc/webkit_sdk/WebKit/WebKitEncoder.mm:
(webrtc::setH264HardwareEncoderAllowed): Deleted.
(webrtc::isH264HardwareEncoderAllowed): Deleted.
* Source/ThirdParty/libwebrtc/Source/webrtc/webkit_sdk/WebKit/WebKitUtilities.h:
* Source/ThirdParty/libwebrtc/Source/webrtc/webkit_sdk/objc/components/video_codec/RTCVideoEncoderH264.mm:
(-[RTCVideoEncoderH264 resetCompressionSessionWithPixelFormat:]):
* Source/WebCore/platform/mediastream/WebRTCProvider.cpp:
(WebCore::WebRTCProvider::webRTCAvailable):
(WebCore::WebRTCProvider::setH264HardwareEncoderAllowed): Deleted.
* Source/WebCore/platform/mediastream/WebRTCProvider.h:
* Source/WebCore/platform/mediastream/gstreamer/GStreamerWebRTCProvider.cpp:
(WebCore::WebRTCProvider::setH264HardwareEncoderAllowed): Deleted.
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProvider.cpp:
(WebCore::WebRTCProvider::setH264HardwareEncoderAllowed): Deleted.
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProviderCocoa.cpp:
(WebCore::WebRTCProvider::setH264HardwareEncoderAllowed): Deleted.
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::resetToConsistentState):
(WebCore::Internals::setH264HardwareEncoderAllowed): Deleted.
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/Internals.idl:

Canonical link: <a href="https://commits.webkit.org/291620@main">https://commits.webkit.org/291620@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/83746fae4565d4a3961914e26e1a37d79c82b6af

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/93482 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/13062 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/2783 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/98484 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/44008 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/95532 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/13338 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/21498 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/71419 "Found 24 new test failures: imported/w3c/web-platform-tests/css/css-position/hypothetical-box-scroll-parent.html imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-auto-last-word-001.html imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-character.html imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-vs-float-clearance-002.html imported/w3c/web-platform-tests/css/css-transforms/animation/transform-non-invertible-discrete-interpolation.html imported/w3c/web-platform-tests/css/css-transforms/transform-box/cssbox-content-box-002.html imported/w3c/web-platform-tests/css/css-view-transitions/massive-element-left-of-viewport-offscreen-new.html imported/w3c/web-platform-tests/css/css-view-transitions/massive-element-on-top-of-viewport-offscreen-new.html imported/w3c/web-platform-tests/css/css-view-transitions/pseudo-with-classes-exit.html imported/w3c/web-platform-tests/css/css-view-transitions/pseudo-with-classes-match-multiple-wildcard.html ... (failure)") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/28798 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/96484 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/9976 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/84534 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/51753 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/9658 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/2158 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/43322 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/79937 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/2206 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/100516 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/20534 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/15013 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/80432 "Found 1 new test failure: imported/w3c/web-platform-tests/service-workers/service-worker/redirected-response.https.html (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/20786 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/80465 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/79762 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/19827 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/24295 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/1640 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/13669 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/20518 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/25696 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/20205 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/23665 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/21946 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

philn reviewed (approved):
r=me (if EWS is green)

youennf committed:
REGRESSION(Sequoia): [ macOS Sequoia wk2 Debug x86_64 ] webrtc/captureCanvas-webrtc-software-h264-high.html is a constant failure.
rdar://145695939
https://bugs.webkit.org/show_bug.cgi?id=288655

Reviewed by Philippe Normand.

We had bugs related to software H264 encoding on intel macOS and added related tests.
This test infrastructure was only available if H264 encoding was done within WebProcess, which is now an unsupported configuration.
We remove the test and corresponding obsolete test infrastructure.

* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/mac-site-isolation/TestExpectations:
* LayoutTests/platform/mac-wk2/TestExpectations:
* LayoutTests/platform/mac/TestExpectations:
* LayoutTests/platform/mac/webrtc/captureCanvas-webrtc-software-encoder-expected.txt: Removed.
* LayoutTests/platform/mac/webrtc/captureCanvas-webrtc-software-encoder.html: Removed.
* LayoutTests/platform/wpe/TestExpectations:
* LayoutTests/webrtc/captureCanvas-webrtc-software-h264-baseline-expected.txt: Removed.
* LayoutTests/webrtc/captureCanvas-webrtc-software-h264-baseline.html: Removed.
* LayoutTests/webrtc/captureCanvas-webrtc-software-h264-high-expected.txt: Removed.
* LayoutTests/webrtc/captureCanvas-webrtc-software-h264-high.html: Removed.
* Source/ThirdParty/libwebrtc/Configurations/libwebrtc.exp:
* Source/ThirdParty/libwebrtc/Source/webrtc/webkit_sdk/WebKit/WebKitEncoder.mm:
(webrtc::setH264HardwareEncoderAllowed): Deleted.
(webrtc::isH264HardwareEncoderAllowed): Deleted.
* Source/ThirdParty/libwebrtc/Source/webrtc/webkit_sdk/WebKit/WebKitUtilities.h:
* Source/ThirdParty/libwebrtc/Source/webrtc/webkit_sdk/objc/components/video_codec/RTCVideoEncoderH264.mm:
(-[RTCVideoEncoderH264 resetCompressionSessionWithPixelFormat:]):
* Source/WebCore/platform/mediastream/WebRTCProvider.cpp:
(WebCore::WebRTCProvider::webRTCAvailable):
(WebCore::WebRTCProvider::setH264HardwareEncoderAllowed): Deleted.
* Source/WebCore/platform/mediastream/WebRTCProvider.h:
* Source/WebCore/platform/mediastream/gstreamer/GStreamerWebRTCProvider.cpp:
(WebCore::WebRTCProvider::setH264HardwareEncoderAllowed): Deleted.
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProvider.cpp:
(WebCore::WebRTCProvider::setH264HardwareEncoderAllowed): Deleted.
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProviderCocoa.cpp:
(WebCore::WebRTCProvider::setH264HardwareEncoderAllowed): Deleted.
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::resetToConsistentState):
(WebCore::Internals::setH264HardwareEncoderAllowed): Deleted.
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/Internals.idl:

Canonical link: https://commits.webkit.org/291620@main
